tests/Unit/StateReconstitutionTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+use Thunk\Verbs\Attributes\Autodiscovery\StateId;
+use Thunk\Verbs\Facades\Verbs;
+use Thunk\Verbs\Lifecycle\StateManager;
+use Thunk\Verbs\Models\VerbSnapshot;
+use Thunk\Verbs\State;
+
+/*
+ * The Problem(s)
+ *
+ * FIRST PROBLEM:
+ * - We try to load state1, but we don't have an up-to-date snapshot
+ * - StateManager::load tries to reconstitute state from events
+ * - One of those Event::apply methods load state2
+ * - Best case scenario: we reconstitute state2 before continuing
+ * - Worst case scenario: reconstituting state2 tries to reconstitute state1, and we're in an infinite loop
+ * - (if no loop) state1 continues to reconstitute, but it's acting with state2 FULLY up-to-date, not
+ *   just up-to-date with where state1 happens to be
+ *
+ * TO TEST FIRST PROBLEM:
+ * - Event1 adds State1::counter to State2::counter and increments State2::counter
+ * - Event2 increments State2::counter
+ *
+ * SECOND PROBLEM:
+ * - We try to load state1, but we don't have an up-to-date snapshot
+ * - StateManager::load tries to reconstitute state from events
+ * - One of those Event::apply methods requires state1 and state2, so we need to load state2
+ * - Reconstituting state2 re-runs the same apply method on state2 before also running it on state1
+ * - Double-apply happens
+ */
```

Comment:
I think we should probably start here and review whether these are actually the problems and if so confirm that the tests effectively test them.